src/statusicon.cpp

```diff
@@ -48,34 +63,62 @@ void StatusIcon::activateAnimation() {
 
 void StatusIcon::animateIcon() {
   Q_ASSERT(m_animatedIconIndex < ANIMATED_LOGO_STEPS.size());
-  setIcon(ANIMATED_LOGO_STEPS[m_animatedIconIndex++]);
+  setIcon(ANIMATED_LOGO_STEPS[m_animatedIconIndex++], false);
   if (m_animatedIconIndex == ANIMATED_LOGO_STEPS.size()) {
     m_animatedIconIndex = 0;
   }
 }
 
+void StatusIcon::stabilityChanged() {
+  logger.debug() << "Stability changed";
+
+  switch (MozillaVPN::instance()->connectionHealth()->stability()) {
+    case ConnectionHealth::Stable:
+      setIndicatorColor(QColor(63, 225, 176, 255));  // green
+      break;
+    case ConnectionHealth::Unstable:
+      setIndicatorColor(QColor(255, 164, 54, 255));  // orange
+      break;
+    case ConnectionHealth::NoSignal:
+      setIndicatorColor(QColor(226, 40, 80, 255));  // red
+      break;
+    default:
+      logger.error() << "Unhandled status indicator for connection stability";
+      setIndicatorColor(QColor());
+  }
+}
+
 void StatusIcon::stateChanged() {
-  logger.debug() << "Show notification";
+  logger.debug() << "State changed";
 
   m_animatedIconTimer.stop();
 
   MozillaVPN* vpn = MozillaVPN::instance();
+  Q_ASSERT(vpn);
 
   // If we are in a non-main state, we don't need to show special icons.
   if (vpn->state() != MozillaVPN::StateMain) {
-    setIcon(LOGO_GENERIC);
+    setIcon(LOGO_GENERIC, false);
     return;
   }
 
   switch (vpn->controller()->state()) {
+#if defined(MVPN_LINUX) || defined(MVPN_WINDOWS)
```

Comment:
wondering if this should be moved to some subclasses instead of this #if/def

Reply:
I was not sure if it’s really necessary here. Since I don’t have a strong preference I left I like this, but I’m also happy to subclass.